src/lando/main/auth.py

```diff
@@ -41,3 +48,84 @@ def create_user(self, claims: dict) -> User:
     def update_user(self, user: User, claims: dict) -> User:
         self.post_auth_hook(user, claims)
         return super().update_user(user, claims)
+
+
+def require_authenticated_user(f):
+    """
+    Decorator which requires a user to be authenticated.
+
+    Raises a PermissionError if a request is by an unauthenticated user.
+    """
+
+    def wrapper(request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            raise PermissionError
+        return f(request, *args, **kwargs)
+
+    return wrapper
+
+
+class require_permission:
+    """
+    Decorator that raises a PermissionError if a user is missing the given permission.
+    """
+
+    def __init__(self, permission):
+        self.required_permission = permission
+
+    def __call__(self, f: Callable) -> Callable:
+        def wrapper(request, *args, **kwargs):
+            if not request.user.has_perm(f"main.{self.required_permission}"):
```

Comment:
We should check for expired permissions here and indicate them in the exception message (assuming those eventually make it to the user).

Reply:
Can we consolidate this discussion in [bug 1908702](https://bugzilla.mozilla.org/show_bug.cgi?id=1908702)?